flink-yarn/src/main/java/org/apache/flink/yarn/YarnApplicationFileUploader.java

```diff
@@ -257,20 +254,15 @@ YarnLocalResourceDescriptor registerSingleLocalResource(
                 if (file.isDirectory()) {
                     final java.nio.file.Path shipPath = file.toPath();
                     final java.nio.file.Path parentPath = shipPath.getParent();
-                    Files.walkFileTree(
-                            shipPath,
-                            new SimpleFileVisitor<java.nio.file.Path>() {
-                                @Override
-                                public FileVisitResult visitFile(
-                                        java.nio.file.Path file, BasicFileAttributes attrs) {
-                                    localPaths.add(new Path(file.toUri()));
-                                    relativePaths.add(
-                                            new Path(
-                                                    localResourcesDirectory,
-                                                    parentPath.relativize(file).toString()));
-                                    return FileVisitResult.CONTINUE;
-                                }
-                            });
+                    Collection<java.nio.file.Path> paths =
+                            FileUtils.listFilesInDirectory(shipPath, path -> true);
```

Comment:
Can you add a test for `Fileutils.listFilesInDirectory` that ensures that it follows symlinks?

Reply:
It's done, please have a look